pkg/build/registry/buildlog/rest.go

```diff
@@ -52,16 +52,16 @@ func (r *REST) ResourceLocation(ctx kapi.Context, id string) (string, error) {
 
 	// TODO: these must be status errors, not field errors
 	// TODO: choose a more appropriate "try again later" status code, like 202
-	if len(build.PodName) == 0 {
-		return "", errors.NewFieldRequired("Build.PodName", build.PodName)
+	if build.PodRef == nil || len(build.PodRef.Name) == 0 || len(build.PodRef.Namespace) == 0 {
+		return "", errors.NewFieldRequired("Build.PodRef", build.PodRef)
```

Comment:
`Build.podRef` - fixed. But I can't extend build's 'validate' since upon creation `Build` does not have `PodRef` filled. It's done only after new Build reaches build controller (see [controller.go](https://github.com/openshift/origin/blob/master/pkg/build/controller/controller.go#L115)). Thus we have it here, the only thing that comes to my mind is that this check is too verbose and I can cut it just to checking `PodRef` against nil.